internal/proxy/connection/manager.go

```diff
@@ -52,11 +53,27 @@ func (s *connectionManager) checkLoop() {
 			return
 		case <-t.C:
 			s.removeLongInactiveClients()
+			// not sure if we should purge them periodically.
+			s.purgeIfNumOfClientsExceed()
 			t.Reset(paramtable.Get().ProxyCfg.ConnectionCheckIntervalSeconds.GetAsDuration(time.Second))
 		}
 	}
 }
 
+func (s *connectionManager) purgeIfNumOfClientsExceed() {
+	if s.count.Load() >= paramtable.Get().ProxyCfg.MaxConnectionNum.GetAsInt64() {
```

Comment:
We have encapsulated the Len interface for typeutil.ConcurrentMap. So, is it unnecessary to have s.count here?
If so, feel free to remove it（s.count）.

Reply:
fixed.